### PR TITLE
[ Layer ] More Parallelization along batch direction

### PR DESCRIPTION
**Pull request by jijoongmoon**
This patch includes more parallelization along batch direction. 
 - Forward function of Pooling Layer
 - calGradient function of LSTM Layer

### CODE REVIEW

**Review by DonghakPark — APPROVED** [review]
LGTM!

File: Applications/VGG/jni/main.cpp [hunk 1]
```diff
@@ -20,6 +20,9 @@
 
 #include <cifar_dataloader.h>
 #include <model.h>
+#ifdef PROFILE
```
Comment (jihochu) [inline Applications/VGG/jni/main.cpp:23]: it doesn't need to check PROFILE

File: Applications/VGG/jni/main.cpp [hunk 2]
```diff
@@ -183,5 +191,9 @@ int main(int argc, char *argv[]) {
     return 1;
   }
 
+#ifdef PROFILE
+  std::cout << *listener;
```
Comment (jihochu) [inline Applications/VGG/jni/main.cpp:195]: How about using #1978  method ?

Comment (jijoongmoon) [reply Applications/VGG/jni/main.cpp:195]: This is same commit which merged. Once the #1978 is merged, I will rebase this.

File: nntrainer/layers/conv2d_layer.cpp [hunk 3]
```diff
@@ -465,7 +466,7 @@ void Conv2DLayer::calcDerivative(RunLayerContext &context) {
   /// col2im(column matrix) to reconstruct the original image
```
Comment (songgot) [inline nntrainer/layers/conv2d_layer.cpp:466]:
```suggestion
/// filter_kernel^T X derivative -> column matrix
```

Comment (songgot) [inline nntrainer/layers/conv2d_layer.cpp:466]: typo: derivaitive -> derivative

**Review by songgot — APPROVED** [review]

**Pull request merged**